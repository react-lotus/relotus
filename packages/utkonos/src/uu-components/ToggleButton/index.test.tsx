import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';

import { ToggleButton } from '.';

describe('ToggleButton', () => {
  it('render', () => {
    const component = render(
      <ToggleButton
        value="value1"
        options={[
          { label: 'label1', value: 'value1' },
          { label: 'label2', value: 'value2' },
        ]}
      />,
    );

    expect(component.container.firstChild).toMatchSnapshot();
  });

  it('has right styles when disabled', () => {
    render(
      <ToggleButton
        disabled
        value="value1"
        options={[
          { label: 'label1', value: 'value1' },
          { label: 'label2', value: 'value2' },
        ]}
      />,
    );

    const button = screen.getByText('label1');

    expect(button).toHaveClass('uu-toggleButtonLabel_disabled');
  });

  it('should not to be clickable when disabled', () => {
    const onChange = jest.fn();

    render(
      <ToggleButton
        disabled
        onChange={onChange}
        value="value2"
        options={[
          { label: 'label1', value: 'value1' },
          { label: 'label2', value: 'value2' },
        ]}
      />,
    );

    const button = screen.getByText('label1');

    fireEvent.click(button);

    expect(onChange).not.toHaveBeenCalled();
  });
});
