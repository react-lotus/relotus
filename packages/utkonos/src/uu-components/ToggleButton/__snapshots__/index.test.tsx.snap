// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleButton render 1`] = `
<div
  class="uu-toggleButton"
>
  <label
    aria-disabled="false"
    class="uu-toggleButtonLabel uu-toggleButtonLabel_checked"
    for="nanoid_value1"
  >
    <input
      checked=""
      class="uu-toggleButtonInput"
      id="nanoid_value1"
      name="nanoid"
      type="radio"
      value="value1"
    />
    label1
  </label>
  <label
    aria-disabled="false"
    class="uu-toggleButtonLabel"
    for="nanoid_value2"
  >
    <input
      class="uu-toggleButtonInput"
      id="nanoid_value2"
      name="nanoid"
      type="radio"
      value="value2"
    />
    label2
  </label>
</div>
`;
